pkgs/development/libraries/qt-5/modules/qtbase.nix

```diff
@@ -34,18 +34,41 @@ let
   compareVersion = v: builtins.compareVersions version v;
   qmakeCacheName = if compareVersion "5.12.4" < 0 then ".qmake.cache" else ".qmake.stash";
   debugSymbols = debug || developerBuild;
-in
 
-stdenv.mkDerivation {
+  libcxxIncDir = "${llvmPackages_10.libcxx.dev}/include/c++/v1";
+  libcxxLibDir = "${llvmPackages_10.libcxx.out}/lib";
+  clangCxxFlags = "-nostdinc++ -isystem ${libcxxIncDir}";
+  clangCxxLdflags = "-nostdlib++ -L ${libcxxIncDir} -Wl,-rpath,${libcxxLibDir} -lc++ -lc++abi";
+
+  # Building with libc++ implies that all statically linked c++ deps are also
```

Comment:
maybe you want to build pkgs.pkgsStatic.pkgsLLVM.qtbase then?
can you add a comment as to why you are using clang instead of the default compiler in stdenv? Doesn't that force all dependent apps to also compile with clang?